src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    private teammates.storage.sqlentity.Section createSection(teammates.storage.sqlentity.Course newCourse,
+            String sectionName) {
+        return new teammates.storage.sqlentity.Section(newCourse, sectionName);
+    }
+
+    private teammates.storage.sqlentity.Team createTeam(teammates.storage.sqlentity.Section section, String teamName) {
+        return new teammates.storage.sqlentity.Team(section, teamName);
+    }
+
+    private void migrateCourseEntity(teammates.storage.sqlentity.Course newCourse) {
+        Map<String, teammates.storage.sqlentity.Section> sectionNameToSectionMap = migrateSectionChain(newCourse);
+        System.out.println(sectionNameToSectionMap); // To stop lint from complaining
+        // TODO: Add mirgrateFeedbackChain
+        // migrateFeedbackChain(sectionNameToSectionMap);
+    }
+
+    private Map<String, teammates.storage.sqlentity.Section> migrateSectionChain(
+            teammates.storage.sqlentity.Course newCourse) {
+        List<CourseStudent> oldStudents = ofy().load().type(CourseStudent.class).filter("courseId", newCourse.getId())
+                .list();
+        Map<String, teammates.storage.sqlentity.Section> sections = new HashMap<>();
+        Map<String, List<CourseStudent>> sectionToStuMap = oldStudents.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getSectionName));
+
+        for (Map.Entry<String, List<CourseStudent>> entry : sectionToStuMap.entrySet()) {
+            String sectionName = entry.getKey();
+            List<CourseStudent> stuList = entry.getValue();
+            teammates.storage.sqlentity.Section newSection = createSection(newCourse, sectionName);
+            sections.put(sectionName, newSection);
+            migrateTeams(newCourse, newSection, stuList);
+            saveEntityDeferred(newSection);
+        }
+        return sections;
+    }
+
+    private void migrateTeams(teammates.storage.sqlentity.Course newCourse,
+            teammates.storage.sqlentity.Section newSection, List<CourseStudent> studentsInSection) {
+        Map<String, List<CourseStudent>> teamNameToStuMap = studentsInSection.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getTeamName));
+        for (Map.Entry<String, List<CourseStudent>> entry : teamNameToStuMap.entrySet()) {
+            String teamName = entry.getKey();
+            List<CourseStudent> stuList = entry.getValue();
+            teammates.storage.sqlentity.Team newTeam = createTeam(newSection, teamName);
+            migrateStudent(newCourse, newTeam, stuList);
+            saveEntityDeferred(newTeam);
+        }
+    }
+
+    private void migrateStudent(teammates.storage.sqlentity.Course newCourse, teammates.storage.sqlentity.Team newTeam,
+            List<CourseStudent> studentsInTeam) {
+        for (CourseStudent oldStudent : studentsInTeam) {
+            teammates.storage.sqlentity.Student newStudent = new teammates.storage.sqlentity.Student(newCourse,
```

Comment:
Should we standardize with the `create..` methods too?

Reply:
Good catch will do!